### DISCUSSION TITLE
cannot select text in simple terminal with 1.4.34 to latest version

### DISCUSSION THREAD

**Discussion by cstkingkey** [fltk-rs/fltk-rs#1613] [General]
with multiple pages of content in the simple terminal, when selecting the text, it will jump to the last page.

**Comment by MoAlyousef** [comment]
Hi
Do you have a minimal repro?

**Comment by cstkingkey** [comment]
when click the content, it will  scroll to the bottom

**Reply by MoAlyousef** [reply]
Can you try replacing it with `terminal::Terminal`. The old SimpleTerminal had several issues and was removed from FLTK.
```rust
use fltk::{prelude::*, *};

fn main() {
    let readme = std::fs::read_to_string("README.md").unwrap();
    let app = app::App::default();
    let mut win = window::Window::new(300, 100, 700, 800, "Hello from Rust");

    let mut term = terminal::Terminal::default_fill();
    term.set_history_rows(1000);
    term.append(&readme);

    win.end();
    win.show();
  
    app.run().unwrap();
}
```

**Reply by cstkingkey** [reply]
can not select the content using mouse

**Reply by cstkingkey** [reply]
how to clear the content?  clear function is not working

**Reply by cstkingkey** [reply]
I can select the text. But it has no right-click menu and I change the color, so it has no visual indicator. After I set the selection color, it works for me. But I can not  find the right function to clear the content like the simpleterminal clear function.

**Reply by cstkingkey** [reply]
still not working in my code. 
And it't not displaying content correctly under linux arm64. the chinese characters and ansi characters are not at the same horizontal level.

**Reply by MoAlyousef** [reply]
Can you try with master using SimpleTerminal. You will need to set `term.set_stay_at_bottom(false);`

**Comment by cstkingkey** [comment]
I cannot copy the text with following lines added.
let widget_scheme = WidgetScheme::new(SchemeType::Fluent);
widget_scheme.apply();

**Reply by MoAlyousef** [reply]
Do you have a repro?
I can't repro on my side

**Reply by cstkingkey** [reply]
```
use fltk::{enums::Color, prelude::*, *};
use fltk_theme::{SchemeType, WidgetScheme};

fn main() {
    let widget_scheme = WidgetScheme::new(SchemeType::Fluent);
    widget_scheme.apply();
    let readme = std::fs::read_to_string("README.md").unwrap();
    let app = app::App::default();
    let mut win = window::Window::new(300, 100, 700, 800, "Hello from Rust");

    let mut term = terminal::Terminal::new(0, 0, 700, 800, "Hello from Rust");
    term.set_history_rows(1000);
    term.set_color(Color::White);
    term.set_text_color(Color::Black);
    term.set_selection_bg_color(Color::Yellow);
    term.append(&readme);
    win.end();
    win.show();
  
    app.run().unwrap();
}
```

**Reply by MoAlyousef** [reply]
By default, Terminal doesn’t show a menu on right click. Can you try with master’s SimpleTerminal?

**Reply by cstkingkey** [reply]
no method named `set_stay_at_bottom` found for struct `Terminal` in the current scope

**Reply by cstkingkey** [reply]
without the line, it's still not working. I cannot copy the content.

**Reply by MoAlyousef** [reply]
SimpleTerminal not Terminal

**Reply by cstkingkey** [reply]
With master code, simpleterminal works and terminal does not.

**Reply by MoAlyousef** [reply]
As explained, Terminal intentionally doesn’t offer a popup.

**Reply by cstkingkey** [reply]
copying with ctrl c not working

**Reply by MoAlyousef** [reply]
Some schemes and all themes if I remember correctly set_visible_focus to false. This means for some widgets like Terminal which depend on visible focus for some operations to ignore them.
You can bring back support for ctrl+c and even get a popup on left click using a menu button:
```rust
use fltk::{prelude::*, *, enums::*};

fn main() {
    let a = app::App::default();
    app::set_visible_focus(false);
    let mut w = window::Window::default().with_size(400, 300);
    let mut term = terminal::Terminal::default_fill();
    term.append("hello world");
    let mut mb = menu::MenuButton::default_fill().with_type(menu::MenuButtonType::Popup3);
    w.end();
    w.show();
    mb.add("Copy", Shortcut::Ctrl | 'c', menu::MenuFlag::Normal, move |_| {
        app::copy(&term.selection_text());
    });
    a.run().unwrap();
}
```
